components/wifi/esp8266-driver/ESP8266Interface.cpp

```diff
@@ -167,9 +179,35 @@ void ESP8266Interface::_oob2global_event_queue()
     }
 }
 
+void ESP8266Interface::_connect_async()
+{
+    _cmutex.lock();
+    if (!_connect_event_id) {
+        tr_debug("_connect_async(): cancelled");
+        _cmutex.unlock();
+        return;
+    }
+
+    if (_esp.connect(ap_ssid, ap_pass) != NSAPI_ERROR_OK) {
+        _connect_event_id = _global_event_queue->call(callback(this, &ESP8266Interface::_connect_async));
```

Comment:
As you will potentially be delaying the global event queue quite a long time here (15 second timeout), at least mitigate it by making this call again after the same timeout or longer. Don't take 100% of the thread time.

Reply:
Fixed